File: DXDAL/ProductionProjectDetails.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Namespace SPGData
    Public Class ProductionProjectDetails
        Inherits AuditableBase

        Private _ProjectDetail As ProjectDetails
        Private _Production As Inventory
        Private _unitsProduced As Integer
        Private _bagsProduced As Integer
        Private _palletsProduced As Integer
        Private _IsOverrun As Boolean

        <Association("Production-ProjectDetail")>
        Public Property ProjectDetail As ProjectDetails
            Get
                Return _ProjectDetail
            End Get
            Set(value As ProjectDetails)
                SetPropertyValue(Of ProjectDetails)("ProjectDetail", _ProjectDetail, value)
            End Set
        End Property

        <Association("ProjectDetails-Production")>
        Public Property Production As Inventory
            Get
                Return _Production
            End Get
            Set(value As Inventory)
                SetPropertyValue(Of Inventory)("Production", _Production, value)
            End Set
        End Property

        Public Property UnitsProduced As Integer
            Get
                Return _unitsProduced
            End Get
            Set(ByVal Value As Integer)
                SetPropertyValue(Of Integer)("UnitsProduced", _unitsProduced, Value)
            End Set
        End Property

        Public Property BagsProduced As Integer
            Get
                Return _bagsProduced
            End Get
            Set(ByVal Value As Integer)
                SetPropertyValue(Of Integer)("BagsProduced", _bagsProduced, Value)
            End Set
        End Property

        Public Property PalletsProduced As Integer
            Get
                Return _palletsProduced
            End Get
            Set(ByVal Value As Integer)
                SetPropertyValue(Of Integer)("PalletsProduced", _palletsProduced, Value)
            End Set
        End Property

        Public Property IsOverrun As Boolean
            Get
                Return _IsOverrun
            End Get
            Set(ByVal Value As Boolean)
                SetPropertyValue(Of Boolean)("IsOverrun", _IsOverrun, Value)
            End Set
        End Property

        Public Sub New()
            MyBase.New()
            ' This constructor is used when an object is loaded from a persistent storage.
            ' Do not place any code here.			
        End Sub

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
            ' This constructor is used when an object is loaded from a persistent storage.
            ' Do not place any code here.			
        End Sub

        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
            ' Place here your initialization code.
        End Sub

        Public Shadows Class FieldsClass
            Inherits DevExpress.Xpo.PersistentBase.FieldsClass
            Public Sub New()
                MyBase.New()
            End Sub
            Public Sub New(ByVal propertyName As String)
                MyBase.New(propertyName)
            End Sub
            Public ReadOnly Property Oid() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Oid"))
                End Get
            End Property
            Public ReadOnly Property ProjectDetail As ProjectDetails.FieldsClass
                Get
                    Return New ProjectDetails.FieldsClass(GetNestedName("ProjectDetail"))
                End Get
            End Property
            Public ReadOnly Property Production As Inventory.FieldsClass
                Get
                    Return New Inventory.FieldsClass(GetNestedName("Production"))
                End Get
            End Property
            Public ReadOnly Property UnitsProduced As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("UnitsProduced"))
                End Get
            End Property
            Public ReadOnly Property BagsProduced As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("BagsProduced"))
                End Get
            End Property
            Public ReadOnly Property PalletsProduced As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("PalletsProduced"))
                End Get
            End Property
            Public ReadOnly Property IsOverrun As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("PalletsProduced"))
                End Get
            End Property
        End Class
        Private Shared _fields As FieldsClass
        Public Shared Shadows ReadOnly Property Fields() As FieldsClass
            Get
                If ReferenceEquals(_fields, Nothing) Then
                    _fields = New FieldsClass()
                End If
                Return _fields
            End Get
        End Property

    End Class

End Namespace

File: DXDAL/ProductionProjectDetails.cs
using DevExpress.Xpo;
using DevExpress.Data.Filtering;

namespace DXDAL
{
    namespace SPGData
    {
        public class ProductionProjectDetails : AuditableBase
        {
            private ProjectDetails _ProjectDetail;
            private Inventory _Production;
            private int _unitsProduced;
            private int _bagsProduced;
            private int _palletsProduced;
            private bool _IsOverrun;

            [Association("Production-ProjectDetail")]
            public ProjectDetails ProjectDetail
            {
                get => _ProjectDetail;
                set => SetPropertyValue("ProjectDetail", ref _ProjectDetail, value);
            }

            [Association("ProjectDetails-Production")]
            public Inventory Production
            {
                get => _Production;
                set => SetPropertyValue("Production", ref _Production, value);
            }

            public int UnitsProduced
            {
                get => _unitsProduced;
                set => SetPropertyValue<int>("UnitsProduced", ref _unitsProduced, value);
            }

            public int BagsProduced
            {
                get => _bagsProduced;
                set => SetPropertyValue<int>("BagsProduced", ref _bagsProduced, value);
            }

            public int PalletsProduced
            {
                get => _palletsProduced;
                set => SetPropertyValue<int>("PalletsProduced", ref _palletsProduced, value);
            }

            public bool IsOverrun
            {
                get => _IsOverrun;
                set => SetPropertyValue("IsOverrun", ref _IsOverrun, value);
            }

            public ProductionProjectDetails() : base()
            {
            }

            public ProductionProjectDetails(Session session) : base(session)
            {
            }

            public override void AfterConstruction()
            {
                base.AfterConstruction();
            }

            public new class FieldsClass : PersistentBase.FieldsClass
            {
                public FieldsClass() : base()
                {
                }
                public FieldsClass(string propertyName) : base(propertyName)
                {
                }
                public OperandProperty Oid => new OperandProperty(GetNestedName("Oid"));
                public ProjectDetails.FieldsClass ProjectDetail => new ProjectDetails.FieldsClass(GetNestedName("ProjectDetail"));
                public Inventory.FieldsClass Production => new Inventory.FieldsClass(GetNestedName("Production"));
                public OperandProperty UnitsProduced => new OperandProperty(GetNestedName("UnitsProduced"));
                public OperandProperty BagsProduced => new OperandProperty(GetNestedName("BagsProduced"));
                public OperandProperty PalletsProduced => new OperandProperty(GetNestedName("PalletsProduced"));
                public OperandProperty IsOverrun => new OperandProperty(GetNestedName("PalletsProduced"));
            }
            private static FieldsClass _fields;
            public static new FieldsClass Fields
            {
                get
                {
                    if (ReferenceEquals(_fields, null))
                        _fields = new FieldsClass();
                    return _fields;
                }
            }
        }
    }
}
